### PR TITLE
Implement further missing opcodes

### PR DESCRIPTION
**Pull request by iamrecursion**
# Summary

This adds a whole host of the ones that are quite simple to manage, expanding our opcode coverage significantly, as well as implements Phi, one of the more complex opcodes that needs to be implemented.

# Details

Please have a detailed think through the `phi` implementation in particular, and tell me if it makes sense.

# Checklist

- [x] Code is formatted by Rustfmt or `scarb fmt`.
- [x] Documentation has been updated if necessary.

### CODE REVIEW

**Review by Eagle941 — APPROVED** [review]
It's clear to me!

File: crates/compiler/src/obj_gen/mod.rs [hunk 1]
```diff
@@ -553,6 +553,302 @@ impl ObjectGenerator {
...
+        // incoming variable, and pick the one corresponding to the block in question.
+        let phi_func = bb.context.add_function(&phi_sig, |bb| -> Result<()> {
+            // We start by getting the first of the parameters, which is the one we need to
+            // compare against. It is safe to index directly as we know it must exist here.
+            let compare_param = &phi_func_params[0];
+
+            // We start by creating the condition variable for each. This variable is either
+            // true or false, and will only be true if the incoming block variable matches
+            // the block identifier for that match arm.
+            let condition_variables = phi_func_params
```
Comment (ktemkin) [inline crates/compiler/src/obj_gen/mod.rs:642]: Honestly, I feel like blocks like this are _less_ clear functionally and more clear with a simpler for, especially where you can do things like comment why you're `continue`'ing on the first variable.,

File: crates/compiler/src/obj_gen/mod.rs [hunk 2]
```diff
@@ -553,6 +553,302 @@ impl ObjectGenerator {
...
+                        Some(&LLVMType::bool),
+                    )?;
+
+                    bb.call_builtin(call_name, parameters, vec![ret_var], Vec::new(), None);
+
+                    Ok(ret_var)
+                })
+                .collect::<Result<Vec<_>>>()?;
+
+            // Unfortunately, FLO requires that match arms jump to blocks and not just
```
Comment (ktemkin) [inline crates/compiler/src/obj_gen/mod.rs:660]: Unfortunately =P? Find me a processor* where you can emit the equivalent code without jump targets for your conditions and I'll consider it unfortunate =P.

* Technically, armv7 can do this with certain instructions -- those instructions have forms like "load if Carry", and you could chain those, but it was so rarely performant and silly that they dropped it in armv8. I'm not counting that.

File: crates/compiler/src/polyfill.rs [hunk 3]
```diff
@@ -1239,6 +1237,6 @@ mod test {
     fn has_correct_polyfill_count() {
         let polyfills = PolyfillMap::new();
         let count = polyfills.iter().count();
-        assert_eq!(count, 842);
+        assert_eq!(count, 952);
```
Comment (ktemkin) [inline crates/compiler/src/polyfill.rs:1240]: I feel like there needs to be a note at the top mentioning to modify this if we add polyfills.

**Review by ktemkin — APPROVED** [review]
You can consider this approved once the CV name is fixed. :)

**Pull request merged**